### PR TITLE
bug/fix resources

### PR DESCRIPTION
**Pull request by Gnzlt**
## What does this implement/fix? Explain your changes.
- Added new scan icon
- Compressed existing resources
- Replaced hardcoded Strings with placeholders
- Added context to new Strings to be translated.

## Any relevant logs, error output, etc?
...

## Any other comments?
...

## Where has this been tested?

- **Devices/Simulators:** 
Google Pixel 4
Google Pixel XL

- **OS Version:** 
Android 11
Android 10

### CODE REVIEW

File: documentscanner/build.gradle [hunk 1]
```diff
@@ -14,6 +14,10 @@ android {
         targetSdkVersion rootProject.ext.compileSdkVersion
         versionCode rootProject.ext.versionCode
         versionName rootProject.ext.versionName
+
+        ndk {
+            abiFilters 'arm64-v8a', 'armeabi-v7a'
```
[the remark this answers is not in the document]
Comment (Gnzlt) [reply documentscanner/build.gradle:19]: As [previously discussed](https://megaconz.slack.com/archives/CBV4ZRC1L/p1598282116010200), we decided to stick to `armeabi-v7a` and `arm64-v8a` due to limitations with APK size.

File: documentscanner/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+    <string name="scan_quality_medium" context="Document scanner medium quality option">Medium</string>
+    <string name="scan_quality_low" context="Document scanner low quality option">Low</string>
+    <string name="scan_dialog_discard_title" context="Title text shown on document scanner dialog to discard scan">Discard scan?</string>
+    <string name="scan_dialog_discard_body" context="Body text shown on document scanner dialog to discard scan">The current scan will be lost</string>
+    <string name="scan_dialog_discard_all_title" context="Title text shown on document scanner dialog to discard scan all scans">Discard all scans?</string>
+    <string name="scan_dialog_discard_all_body" context="Body text shown on document scanner dialog to discard all scans">The current document will be lost</string>
+    <string name="scan_dialog_delete_body" context="Body text shown on document scanner dialog to delete current scan">Delete current scan?</string>
+    <string name="scan_requires_permission" context="Message shown when users deny camera permissions to scan documents">Camera permissions are needed to scan documents</string>
 </resources>
```
Comment (javiergm1983) [inline documentscanner/src/main/res/values/strings.xml:26]: Rename this file by `strings_document_scanner.xml` in order to upload it to Babel. The filename `strings.xml` is already in use in Babel for the strings of the Android app.

File: documentscanner/src/main/res/values/strings_document_scanner.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <string name="scan_quality_low" context="Document scanner low quality option">Low</string>
+    <string name="scan_dialog_discard_title" context="Title text shown on document scanner dialog to discard scan">Discard scan?</string>
+    <string name="scan_dialog_discard_body" context="Body text shown on document scanner dialog to discard scan">The current scan will be lost</string>
+    <string name="scan_dialog_discard_all_title" context="Title text shown on document scanner dialog to discard scan all scans">Discard all scans?</string>
+    <string name="scan_dialog_discard_all_body" context="Body text shown on document scanner dialog to discard all scans">The current document will be lost</string>
+    <string name="scan_dialog_delete_body" context="Body text shown on document scanner dialog to delete current scan">Delete current scan?</string>
+    <string name="scan_requires_permission" context="Message shown when users deny camera permissions to scan documents">Camera permissions are needed to scan documents</string>
+    <string name="scan_invalid_input" context="Error text for field invalid input">Invalid input</string>
+</resources>
```
Comment (javiergm1983) [inline documentscanner/src/main/res/values/strings_document_scanner.xml:26]: Add screenshots for each of these strings to the screenshots shared folder.

**Review by javiergm1983 — CHANGES_REQUESTED** [review]
Some strings improvements needed. Please take a look to my comments.

In addition there is a problem with other string. As you can see in the following screenshot it says "Upload" where it should say "Add" (see the [design](https://projects.invisionapp.com/d/main#/console/13743399/427943818/preview#project_console))

Screenshot_20201019-175719

File: documentscanner/src/main/res/values/strings_document_scanner.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<resources>
+    <string name="app_name">MEGA Document Scanner</string>
```
Comment (javiergm1983) [inline documentscanner/src/main/res/values/strings_document_scanner.xml:2]: Please, also add a context for the app name string. I'm sure that translators wil require it.
For example: "Name of the MEGA Document Scanner. Keep uppercase."

```suggestion
    <string name="app_name" context="Name of the MEGA Document Scanner. Keep uppercase.">MEGA Document Scanner</string>
```

File: documentscanner/src/main/res/values/strings_document_scanner.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <string name="scan_title_save_scan" context="Screen title for saving current scan">Save scan</string>
+    <string name="scan_action_add" context="Action text to add a new page">Add</string>
+    <string name="scan_action_save" context="Action text to save document">Save</string>
+    <string name="scan_action_retake" context="Action text to retake current scan picture">Retake</string>
+    <string name="scan_file_name" context="Document scanner file name">File name</string>
+    <string name="scan_file_type" context="Input field description in the document save screen">File type</string>
+    <string name="scan_quality" context="Input field description in the document save screen">Quality</string>
+    <string name="scan_destination" context="Input field description in the document save screen">Destination</string>
+    <string name="scan_file_type_pdf" context="Document scanner PDF file type option">PDF</string>
+    <string name="scan_file_type_jpg" context="Document scanner JPG file type option">JPG</string>
```
Comment (javiergm1983) [inline documentscanner/src/main/res/values/strings_document_scanner.xml:14]: These two strings are no translatables, so they should not be included in the strings file that we will upload to Babel. They should be converted as constants or moved to a separated file (for example `strings_donottranslate.xml`) like in the MEGA Android app project.

File: documentscanner/src/main/res/values/strings_document_scanner.xml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    <string name="scan_dialog_discard_title" context="Title text shown on document scanner dialog to discard scan">Discard scan?</string>
+    <string name="scan_dialog_discard_body" context="Body text shown on document scanner dialog to discard scan">The current scan will be lost</string>
+    <string name="scan_dialog_discard_all_title" context="Title text shown on document scanner dialog to discard scan all scans">Discard all scans?</string>
+    <string name="scan_dialog_discard_all_body" context="Body text shown on document scanner dialog to discard all scans">The current document will be lost</string>
+    <string name="scan_dialog_delete_body" context="Body text shown on document scanner dialog to delete current scan">Delete current scan?</string>
+    <string name="scan_requires_permission" context="Message shown when users deny camera permissions to scan documents">Camera permissions are needed to scan documents</string>
+    <string name="scan_invalid_input" context="Error text for field invalid input">Invalid input</string>
+    <string name="scan_format_page_count" context="Format text to show current page number out of total">%1$d / %2$d</string>
```
Comment (javiergm1983) [inline documentscanner/src/main/res/values/strings_document_scanner.xml:25]: Same here, this is a no translatable string.

**Review by javiergm1983 — CHANGES_REQUESTED** [review]
I have seen some other problems with the strings. Sorry for not seeing and reporting them yesterday along with the other issues.

**Review by javiergm1983 — APPROVED** [review]

**Comment by javiergm1983** [comment]
I think everything is fine now. @ash-mega , can you take a look at it too?

**Pull request merged**